frontend/apps/reader/modules/readerpagemap.lua

```diff
@@ -327,127 +357,207 @@ function ReaderPageMap:onDocumentRerendered()
     self.page_labels_cache = nil
 end
 
+function ReaderPageMap:applyCharsPerSyntheticPage(spin_widget, value)
+    UIManager:show(ConfirmBox:new{
+        text = _("Apply changes?\nThe document will be reloaded."),
+        ok_callback = function()
+            spin_widget:onClose()
+            self.ui.doc_settings:saveSetting("pagemap_chars_per_synthetic_page", value)
+            if value == nil then
+                self.ui.document:invalidateCacheFile()
+            end
+            local after_open_callback = function(ui)
+                ui.annotation:setNeedsUpdateFlag()
+            end
+            self.ui:switchDocument(self.ui.document.file, nil, after_open_callback)
```

Comment:
I don't think you need to reload the document everytime we change the value.
Just calling buildSynthetic can work on the current document without the need to fully reload/rerender it.
It's only when switching from Synthetic to DocumentProvided (real) that we should really reload.
If we disable synthetic and there is no documentprovided, I think we can just let the synthetic pagemap data be: if we re-enable them, they will be there ready to use - it's like (when not using synthetic at all) real reference page being there even if we don't "use pagemap".

(I guess it's simpler for you to handle it all in the postInit and just use reload - but simpler is not always better :))

Reply:
Without reloading page labels are not updated on changing `chars_per_synthetic_page`.
```lua
self.ui.document:buildSyntheticPageMap(value)
print("LAST", self.ui.document:getPageMapLastPageLabel())
```
```
CRE: building synthetic page map (1000)
LAST	205
CRE: building synthetic page map (2000)
LAST	205
```

Reply:
Ok, so either you call `self.ui.document:resetCallCache()` here in the places where you know pagemap has changed (only in a few tedious places we do that).
Or you add `elseif name == "buildSyntheticPageMap " then add_reset = true` after these (prefered, if it is only always after calling it that things may change - no worry if you call it with the same value and nothing changes, better be simple, things will get recached):
https://github.com/koreader/koreader/blob/7c677bf2ae391bf7f8a43b32ed9fecccb89f978a/frontend/document/credocument.lua#L1803-L1813